webcompat/helpers.py

```diff
@@ -99,61 +99,72 @@ def get_user_info():
         session['avatar_url'] = gh_user.get('avatar_url')
 
 
-def get_browser_name(user_agent_string):
-    '''Return just the browser name.
-
-    unknown user agents will be reported as "other".
-    '''
-    ua_dict = user_agent_parser.Parse(user_agent_string)
-    name = ua_dict.get('user_agent').get('family').lower()
-    device = ua_dict.get('device').get('model')
-    if (name == 'firefox mobile' and
-            ua_dict.get('os').get('family') == 'Firefox OS'):
-        name = 'other'
-    if device == 'Tablet':
-        name += " " + device.lower()
-    return name
-
-
-def get_browser(user_agent_string):
+def get_browser(user_agent_string=None):
     '''Return browser name family and version.
 
     It will pre-populate the bug reporting form.
     '''
-    ua_dict = user_agent_parser.Parse(user_agent_string)
-    ua = ua_dict.get('user_agent')
-    name = ua.get('family')
-    version = ua.get('major', u'Unknown')
-    # Add on the minor and patch version numbers if they exist
-    if version != u'Unknown' and ua.get('minor'):
-        version = version + "." + ua.get('minor')
-        if ua.get('patch'):
-            version = version + "." + ua.get('patch')
-    else:
-        version = ''
-    # Check for tablet devices
-    if ua_dict.get('device').get('model') == 'Tablet':
-        model = ' (Tablet)'
-    else:
-        model = ''
-    return '{0} {1}{2}'.format(name, version, model)
+    if user_agent_string and isinstance(user_agent_string, basestring):
+        ua_dict = user_agent_parser.Parse(user_agent_string)
+        ua = ua_dict.get('user_agent')
+        name = ua.get('family')
```

Comment:
> which would allow you to create a function for a identical piece of code

Disadvantage being I can't re-use `get_browser` for `get_browser_name` anymore.

Rather than re-write everything, I plan on fixing the problems you found and then I'll file a follow-up bug to simplify/make-more-generic these functions. Let's land this to get rid of the 500s and then come back to make it pretty.